modules/packages/LocalAtomics.chpl

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module LocalAtomics {
```

Comment:
It seems to me that this module should not be called LocalAtomics. This will need to be addressed before merging.

I think it's called LocalAtomics because that's how the EBR implementation uses it. But that is not actually corresponding to what is here.

You could:
 a) Consider this part of the EBR implementation and so add it as a submodule to EBR rather than in to modules/packages
 b) Rename it to something more reflective of what it is, for example, `AtomicsABA` or `LocalAtomicObjects` or `AtomicObjects`

Generally speaking I do not understand why this module includes support for wide pointers. But whether or not that makes sense depends on how you choose between a/b above (and if b, what is the name of the module?). Also, if you choose b), you will need to create some module-level chpldoc describing the purpose of this module and giving an example.

I will need to review this module again once you have addressed this feedback in some way.

Reply:
@dgarvit - I recommend making this a sub module for the time being, but the choice is yours for whether it be exposed. I would prefer that it is exposed as people should know about it, but that creates work for you, so I'll leave it as optional.

Reply:
I think I can make a quick patch to rename/revise/refactor and document `LocalAtomics`. I think it'd be nice to have it split into...

1) `LocalAtomicObject` - Without any pointer compression or anything like that, it just discards the locality information. Now has a `param` flag to determine whether or not ABA wrapper will be used, which can eliminate some relatively costly overhead when the ABA features are not utilized.
2) `AtomicObject` - Like `LocalAtomicObject` but with pointer compression.

Reply:
Done. Made a single `AtomicObject` that has two switches that allows it to enable `hasABASupport` and `hasGlobalSupport`; documentation also finished.